File: python/facade/concrete_facade.py
from datetime import datetime
from abstract_facade import AbstractFacade

class ConcreteFacade(AbstractFacade):
    
    """
    This is the concrete implementation of the abstract facade interface.
    """
    _subsystem1 = None
    _subsystem2 = None
    
    def __init__(self, subsystem1, subsystem2):
        self._subsystem1 = subsystem1
        self._subsystem2 = subsystem2
    
    def greet(self):
        name = self._subsystem1.get_name()
        birth = self._subsystem1.get_year_of_birth()
        age = self._subsystem2.compute_age(self._subsystem1, datetime.now().year)
        print "Hello, my name is %s, I was born in %s and therefore I am %s" % \
            (name, str(birth), str(age))
        

File: java/src/tk/csparpa/gofdp/facade/ConcreteFacade.java
package tk.csparpa.gofdp.facade;

import java.util.Calendar;

/**
 * This is the concrete implementation of the abstract façade interface. This
 * class knows the interfaces of the subsystem classes and invokes them
 */
public class ConcreteFacade implements AbstractFacade {

	private AbstractSubsystemModule1 module1;
	private AbstractSubsystemModule2 module2;
	
	public ConcreteFacade(AbstractSubsystemModule1 module1,
						AbstractSubsystemModule2 module2) {
		this.module1 = module1;
		this.module2 = module2;
	}

	@Override
	public void greet() {
		String name = this.module1.getName();
		int birth = this.module1.getYearOfBirth();
		int age = this.module2.computeAge(this.module1, 
									Calendar.getInstance().get(Calendar.YEAR));
		String message = "Hello, my name is " + name + ", I was born in " +
				Integer.toString(birth) + " and therefore I am " + 
				Integer.toString(age);
		System.out.println(message);
	}
}
